projects/RabbitMQ.Client/client/events/EventingBasicConsumer.cs

```diff
@@ -30,16 +30,16 @@
 //---------------------------------------------------------------------------
 
 using System;
+using RabbitMQ.Client.client.impl.Channel;
 
 namespace RabbitMQ.Client.Events
 {
     ///<summary>Experimental class exposing an IBasicConsumer's
     ///methods as separate events.</summary>
     public class EventingBasicConsumer : DefaultBasicConsumer
```

Comment:
Wouldn't it be better to kill all the async stuff and opt for async only?

Reply:
I agree here as well. You could create an async consumer that exposes async events for eventing purposes. The base consumer implementation should i.m.o only have overrideable ValueTasks methods.